File: 1627-last-moment-before-all-ants-fall-out-of-a-plank/1627-last-moment-before-all-ants-fall-out-of-a-plank.py
class Solution:
    def getLastMoment(self, n: int, left: List[int], right: List[int]) -> int:
        leftSteps = max(left) if len(left)>0 else 0
        rightSteps = n-(min(right) if len(right)>0 else n)
        return max(leftSteps,rightSteps)
        

File: 1627-last-moment-before-all-ants-fall-out-of-a-plank/1627-last-moment-before-all-ants-fall-out-of-a-plank.java
class Solution {
    public int getLastMoment(int n, int[] left, int[] right) {
        List<Integer> leftList = getList(left);
        List<Integer> rightList = getList(right);
        int maxLeft = !leftList.isEmpty() ? Collections.max(leftList) : 0;
        int minRight = n - (!rightList.isEmpty() ? Collections.min(rightList) : n);

        return Math.max(maxLeft, minRight);
    }

    public List<Integer> getList(int[] arr) {
        List<Integer> list = new ArrayList<>();
        for (int x : arr) {
            list.add(x);
        }

        return list;

    }
}